src/app/restapi.lua

```diff
@@ -1,266 +1,155 @@
--- ! @function min_temp   print the current temperature
+-- ! @function change config   modify the current config.json file
 --
--- !	@param req  				server request
+-- !	@param req  		      		server request
 -------------------------------------
-function restapi.min_temp_get(req)
-    local body_data = {
-        message = "success",
-        mintemp = restapi.incubator.min_temp
-    }
-
-    local body_json = sjson.encode(body_data)
-
-    return {
-        status = "200 OK",
-        type = "application/json",
-        body = body_json
-    }
-
-end -- end function
-
--- ! @function maxtemp   print the current temperature
---
--- !	@param req  				server request
--------------------------------------
-function restapi.max_temp_post(req)
-    local reqbody = req.getbody()
-    print(reqbody,"max_temp_post")
-    local body_json = sjson.decode(reqbody)
-
-    -- Obtener el nuevo valor de max_temp del cuerpo de la solicitud POST
-    print(body_json.maxtemp)
-    local new_max_temp = body_json.maxtemp
-
-    if type(new_max_temp) == "number" and new_max_temp < 42 and new_max_temp >= 0 and new_max_temp >= restapi.incubator.min_temp then
-
-        restapi.incubator.max_temp = new_max_temp
-
-        return {
-            status = "201 Created"
-        }
-    else
-
-        return {
-            status = "400 Bad Request"
-        }
-    end
-end
-
--- ! @function maxtemp   print the current temperature
---
--- !	@param req  				server request
--------------------------------------
-function restapi.min_temp_post(req)
-
-    local reqbody = req.getbody()
-    print(reqbody)
-
-    local body_json = sjson.decode(reqbody)
-
-    -- Obtener el nuevo valor de max_temp del cuerpo de la solicitud POST
-    print(body_json.mintemp)
-    local new_min_temp = body_json.mintemp
-    if (type(new_min_temp) == "number") then
-        if new_min_temp >= 0 and new_min_temp <= restapi.incubator.max_temp then
-
-            restapi.incubator.min_temp = new_min_temp
-            return {
-                status = "201 Created"
-            }
-
-        else
-            return {
-                status = "400 Bad Request"
-            }
-
-        end
-    end
+function restapi.change_config(req)
+	-- RESPONSES 
+	local error_changing_config =
+	{
+		status = "400 Bad Request",
+		type = "application/json",
+		body = sjson.encode({
+			message =
+			"Error: It cannot be nil or equal to the existing value."
+		})
+	}
+
+	local success_response =
+	{
+		status = "200 OK",
+		type = "application/json",
+		body = sjson.encode({ message = "Json updated successfully" })
+	}
+
+	local error_response =
+	{
+		status = "500 Internal Server Error",
+		type = "application/json",
+		body = sjson.encode({ message = "Failed to update JSON file" })
+	}
+
+	-- Local Variables
+	local request_body_json = req.getbody()
+	local body_table = sjson.decode(request_body_json)
+
+	-- Update the configuration values from the request body
+	if body_table.max_temperature then
```

Comment:
que pasa si no tiene maxtemp el json que viene ?

Reply:
lo deja pasar, pero en el set de la temperatura devuelve false si es igual o nil, entonces retorna mensaje de respuesta:
[![cant-be-equal-or-nil.png](https://i.postimg.cc/8C26mqx9/cant-be-equal-or-nil.png)](https://postimg.cc/LqV5H093)